### ISSUE TITLE
error TS5033: Could not write file EEXIST

### ISSUE THREAD

**Issue by pauldraper** [microsoft/TypeScript#22423]
When /my exists and is empty, and I run one tsc instance with --outDir=/my/output/dir1 and concurrently run another instance with --outDir=/my/output/dir2, I sometimes get an error

```
error TS5033: Could not write file '/my/output/dir2/file.js': EEXIST: file already exists, mkdir '/my/output'.
```

**Comment by pauldraper** [comment]
The solution here is fairly simple; rather than both check if the directory exists and then creating it, just attempt creation and ignore EEXIST. As a bonus, it reduces the number of fs operations from 2 to 1.

**Comment by jakebailey** [comment]
#30135 was a duplicate of this, and was fixed in #30136 by ignoring EEXIST.
